### PR TITLE
Implement 50 move draw to resolve #5

### LINKED ISSUE #5: Cannot claim 50-move rule draw
Self-explanatory. The logic is there for it in `chess.hoon`, but the frontend does not offer players a way to claim the draw.

### PR DESCRIPTION
**Pull request by rovmug-ticfyn**

### CODE REVIEW

**Review by ashelkovnykov — CHANGES_REQUESTED** [review]
Minor, but on lines 529 and 533 in `%chess-debug-inject`, the initial state of `special-draws-available` depends on the FEN being injected (namely on whether the `ply-50-move-rule` is above 100, since it's impossible for an injected position to be a repetition).

File: src/urbit/app/chess.hoon [hunk 1]
```diff
@@ -526,11 +526,11 @@
         ==
       =/  fen  (position-to-fen u.new-position)
       :-  :~  :*  %give  %fact  ~[/game/(scot %da game-id.action)/updates]
-                  %chess-update  !>([%position game-id.action fen |])
+                  %chess-update  !>([%position game-id.action fen (gth ply-50-move-rule.u.new-position 100)])
```
Comment (ashelkovnykov) [inline src/urbit/app/chess.hoon]: Should be `gte`, not `gth`

File: src/urbit/app/chess.hoon [hunk 2]
```diff
@@ -526,11 +526,11 @@
...
               ==
           ==
       %=  this
-        games  (~(put by games) game-id.action [game.action u.new-position *(map @t @ud) | & | | |])
+        games  (~(put by games) game-id.action [game.action u.new-position *(map @t @ud) (gth ply-50-move-rule.u.new-position 100) & | | |])
```
Comment (ashelkovnykov) [inline src/urbit/app/chess.hoon]: Same as above

File: src/urbit/app/chess.hoon [hunk 3]
```diff
@@ -899,12 +899,17 @@
   =/  in-stalemate  ?:  in-checkmate
                       |
                     ~(in-stalemate with-position u.new-position)
-  =/  special-draw-available  (check-threefold new-fen-repetition u.new-position)
+  =/  special-draw-available
+    ?:  ?|  (check-threefold new-fen-repetition u.new-position)
+            (check-50-move-rule u.new-position)
```
Comment (ashelkovnykov) [inline src/urbit/app/chess.hoon]: Syntax error. Should be:
```
?|  (check-threefold new-fen-repetition u.new-position)
    (check-50-move-rule u.new-position)
==
```

**Review by ashelkovnykov — CHANGES_REQUESTED** [review]
Bugs

File: src/urbit/app/chess.hoon [hunk 4]
```diff
@@ -526,11 +526,11 @@
         ==
       =/  fen  (position-to-fen u.new-position)
       :-  :~  :*  %give  %fact  ~[/game/(scot %da game-id.action)/updates]
-                  %chess-update  !>([%position game-id.action fen |])
+                  %chess-update  !>([%position game-id.action fen (gte ply-50-move-rule.u.new-position 100)])
```
Comment (ashelkovnykov) [inline src/urbit/app/chess.hoon]: I just realized we're replicating the `check-50-move-rule` logic - we should be calling `check-50-move-rule` instead:
```
:-  :~  :*  %give  %fact  ~[/game/(scot %da game-id.action)/updates]
            %chess-update  !>([%position game-id.action fen (check-50-move-rule u.new-position)])
```

File: src/urbit/app/chess.hoon [hunk 5]
```diff
@@ -526,11 +526,11 @@
...
               ==
           ==
       %=  this
-        games  (~(put by games) game-id.action [game.action u.new-position *(map @t @ud) | & | | |])
+        games  (~(put by games) game-id.action [game.action u.new-position *(map @t @ud) (gte ply-50-move-rule.u.new-position 100) & | | |])
```
Comment (ashelkovnykov) [inline src/urbit/app/chess.hoon]: Same as above:
```
games (~(put by games) game-id.action [game.action u.new-position *(map @t @ud) (check-50-move-rule u.new-position) & | | |])
```

**Review by ashelkovnykov — APPROVED** [review]
LGTM. Code already correct, but duplicates logic (see comments below). Can be merged after resolution.

**Review by ashelkovnykov — CHANGES_REQUESTED** [review]
Oh, and squash to one commit please

**Comment by rovmug-ticfyn** [comment]
Ready to merge now

**Review by ashelkovnykov — APPROVED** [review]
LGTM

**Pull request merged**